### DISCUSSION TITLE
describe my changes before submitting a pull request

### DISCUSSION THREAD

**Discussion by fxsome** [open-webui/open-webui#2978] [General]
https://github.com/open-webui/open-webui/issues/2949
I Request to Limit Dialog Rounds posted to llm, and [tjbck](https://github.com/tjbck) told me PR Welcome. So I did it myself
when open a pull request , I was told to open a discussion post .

Four ways to submit message to backend:
1,no limit;
2,First N rounds and most recent user message
3,System prompt , first user message and latest N rounds
4,only recent N rounds

I modied 3 files :
src\lib\stores\index.ts  add 2 local parameters 
src\lib\components\chat\Settings\Chats.svelte  deal with the interface to modify the 2 parameters.
src\lib\components\chat\Chat.svelte using getMessagesByStrategy method to apply strategy and apply **only in sendPromptOpenAI method**

**Comment by justinh-rahb** [comment]
You're talking about managing the context. Related:

- #1268

**Comment by fxsome** [comment]
yes ,that's similar.  they seems to talk about tokens. what I made is to limit turns, not tokens.

**Reply by justinh-rahb** [reply]
This from our Pipelines project is related then:
- https://github.com/open-webui/pipelines/blob/main/examples/filters/conversation_turn_limit_filter.py
